lib/PrintAsObjC/PrintAsObjC.cpp

```diff
@@ -479,7 +487,9 @@ class ObjCPrinter : private DeclVisitor<ObjCPrinter>,
 
     // Swift designated initializers are Objective-C designated initializers.
     if (auto ctor = dyn_cast<ConstructorDecl>(AFD)) {
-      if (ctor->isDesignatedInit() &&
+      if (ctor->hasStubImplementation() || ctor->getFormalAccess() < minRequiredAccess) {
```

Comment:
Although…can you add a comment here saying this will only happen if the overridden initializer had the required access?